collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java

```diff
@@ -92,7 +93,36 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
     private void initToolbar() {
         mToolbar = (Toolbar) findViewById(R.id.toolbar);
-        setSupportActionBar(mToolbar);
+        mToolbar.inflateMenu(R.menu.menu);
+
+        mToolbar.findViewById(R.id.menu_sort).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mDrawerLayout.isDrawerOpen(Gravity.END)) {
+                    mDrawerLayout.closeDrawer(Gravity.END);
+                } else {
+                    Collect.getInstance().hideKeyboard(mInputSearch);
+                    mDrawerLayout.openDrawer(Gravity.END);
+                }
+            }
+        });
+
+        mToolbar.findViewById(R.id.menu_filter).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mSearchBoxLayout.getVisibility() == View.GONE) {
+                    showSearchBox();
+                } else {
+                    hideSearchBox();
+                }
+            }
+        });
+
+        boolean hasHardwareMenu =
```

Comment:
Maybe it's but it's always dangerous as this check might doesn't work on some devices... but it's not very important I think so we can leave that and time will tel if it is ok.
Looks like you fixed all bugs I've reported so I'll test everything again tomorrow and let you now if everything works well.